static/app/components/codeSnippet.tsx

```diff
@@ -33,10 +29,27 @@ export function CodeSnippet({
 }: CodeSnippetProps) {
   const ref = useRef<HTMLModElement | null>(null);
 
-  useEffect(
-    () => void (ref.current && Prism.highlightElement(ref.current, false)),
-    [children]
-  );
+  useEffect(() => {
+    const element = ref.current;
+    if (!element) {
+      return;
+    }
+
+    if (language in Prism.languages) {
+      Prism.highlightElement(element);
+      return;
+    }
+
+    import(`prismjs/components/prism-${language}.min`)
+      .then(() => Prism.highlightElement(element))
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.warn(
+          'Prism language not found. Check the `language` prop passed to `<CodeSnippet />`.',
+          error.message
+        );
+      });
+  }, [children, language]);
```

Comment:
You can use `await` with a try catch for this too to make it a bit cleaner.

Reply:
great suggestion! https://github.com/getsentry/sentry/pull/44146/commits/dd89ac012adc713434d7852d5d7ae74407227a4f